site/en/tutorials/load_data/tf-records.ipynb

```diff
@@ -0,0 +1,973 @@
+        "# First, process the two images into tf.Example messages.\n",
+        "# Then, write to a .tfrecords file.\n",
+        "\n",
+        "writer = tf.python_io.TFRecordWriter('images.tfrecords')\n",
+        "\n",
+        "for filename, label in image_labels.items():\n",
+        "  \n",
+        "  file = open(filename, 'rb').read()\n",
+        "\n",
+        "  image_shape = tf.image.decode_jpeg(file).shape\n",
+        "  image_string = base64.b64encode(file)\n",
+        "\n",
+        "  feature = {\n",
+        "      'height': _int64_feature(image_shape[0]),\n",
+        "      'width': _int64_feature(image_shape[1]),\n",
+        "      'depth': _int64_feature(image_shape[2]),\n",
+        "      'label': _int64_feature(label),\n",
+        "      'image_raw': _bytes_feature(image_string),\n",
+        "  }\n",
+        "  \n",
+        "  tfExample = tf.train.Example(features=tf.train.Features(feature=feature))\n",
+        "  writer.write(tfExample.SerializeToString())\n",
+        "\n",
+        "writer.close()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "yJrTe6tHPCfs",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "!ls"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "jJSsCkZLPH6K",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "We now have the file `images.tfrecords`. We can now iterate over the records in the file to read back what we wrote. Since, for our use case we will just reproduce the image, the only feature we need is the raw image string. We can extract that using the getters described above, namely `example.features.feature['image_raw'].bytes_list.value[0]`. We also use the labels to determine which record is the cat as opposed to the bridge."
+      ]
+    },
+    {
+      "metadata": {
+        "colab_type": "code",
+        "id": "M6Cnfd3cTKHN",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "record_iterator = tf.python_io.tf_record_iterator(path='images.tfrecords')\n",
+        "\n",
+        "for string_record in record_iterator:\n",
+        "  example = tf.train.Example()\n",
+        "  example.ParseFromString(string_record)\n",
+        "  \n",
+        "  label = example.features.feature['label'].int64_list.value[0]\n",
+        "  \n",
+        "  # Use the label to determine which image is in each record.\n",
+        "  # The cat image has label 0, and the bridge image has label 1.\n",
+        "\n",
+        "  if label == 0:\n",
+        "    cat_image_string = example.features.feature['image_raw'].bytes_list.value[0]\n",
+        "  if label == 1:\n",
+        "    bridge_image_string = example.features.feature['image_raw'].bytes_list.value[0]"
```

Comment:
This would be cleaner if you created a dict of `images[label] = bytes`. That removes the need to explicitly check for label, and is more scalable in the case that a user has many labels. Below, you can just iterate over the dict/grab the keys directly as necessary.

Reply:
Done